{
  try {
    return future.get();
  }
 catch (  ExecutionException e) {
    throw new RuntimeException(e);
  }
catch (  InterruptedException e) {
    throw new RuntimeException(e);
  }
}
